domains/transfer/Transfer.API/API/Transfer/Startup.cs

```diff
@@ -22,53 +23,26 @@ public static class Startup
-        services.AddOptions<OtlpOptions>().BindConfiguration(OtlpOptions.Prefix).ValidateDataAnnotations().ValidateOnStart();
+        services.AddOptions<ProjectOriginOptions>().BindConfiguration(ProjectOriginOptions.ProjectOrigin)
+            .ValidateDataAnnotations().ValidateOnStart();
+        services.AddOptions<OtlpOptions>().BindConfiguration(OtlpOptions.Prefix).ValidateDataAnnotations()
+            .ValidateOnStart();
 
-        Configuration.Setup()
-            .UseEntityFramework(ef => ef
-                .AuditTypeExplicitMapper(config => config
-                    .Map<TransferAgreement, TransferAgreementHistoryEntry>((evt, eventEntry, historyEntity) =>
-                    {
-                        var actorId = evt.CustomFields.ContainsKey("ActorId") ? evt.CustomFields["ActorId"].ToString() : null;
-                        var actorName = evt.CustomFields.ContainsKey("ActorName") ? evt.CustomFields["ActorName"].ToString() : null;
-
-                        historyEntity.Id = Guid.NewGuid();
-                        historyEntity.CreatedAt = DateTimeOffset.UtcNow;
-                        historyEntity.AuditAction = eventEntry.Action;
-                        historyEntity.ActorId = actorId ?? string.Empty;
-                        historyEntity.ActorName = actorName ?? string.Empty;
-
-                        switch (eventEntry.Action)
-                        {
-                            case "Insert":
-                                historyEntity.TransferAgreementId = (Guid)eventEntry.ColumnValues["Id"];
-                                break;
-                            case "Update":
-                                {
-                                    historyEntity.TransferAgreementId = (Guid)eventEntry.PrimaryKey.Values.First();
-                                    break;
-                                }
-                        }
-                        return true;
-                    })));
 
         var otlpConfiguration = configuration.GetSection(OtlpOptions.Prefix);
         var otlpOptions = otlpConfiguration.Get<OtlpOptions>()!;
 
         services.AddOpenTelemetry()
             .WithMetrics(provider =>
                 provider
-                    .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(TransferAgreementAutomationMetrics.MetricName))
+                    .SetResourceBuilder(ResourceBuilder.CreateDefault()
+                        .AddService(TransferAgreementAutomationMetrics.MetricName))
                     .AddMeter(TransferAgreementAutomationMetrics.MetricName)
                     .AddHttpClientInstrumentation()
                     .AddAspNetCoreInstrumentation()
                     .AddRuntimeInstrumentation()
                     .AddProcessInstrumentation()
                     .AddOtlpExporter(o => o.Endpoint = otlpOptions.ReceiverEndpoint));
```

Comment:
This is not specific for TransferAgreement automation, but applies for the entire application.

Reply:
Moved to program